src/tools/synthetic_node.rs

```diff
@@ -242,8 +242,8 @@ impl SyntheticNode {
     pub async fn wait_for_connection(&self) -> SocketAddr {
         const SLEEP: Duration = Duration::from_millis(10);
         loop {
-            if let Some(addr) = self.known_peers().read().keys().next() {
-                return *addr;
+            if let Some(addr) = self.inner_node.connected_peers().pop() {
```

Comment:
```suggestion
            // Mutating the collection is alright since this is a copy of the connections and not the actualy list.
            if let Some(addr) = self.inner_node.connected_peers().pop() {
```